src/Sylius/Component/Promotion/Generator/Instruction.php

```diff
@@ -15,23 +15,46 @@
  * Coupon generate instruction.
  *
  * @author Paweł Jędrzejewski <pawel@sylius.org>
+ * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
  */
-class Instruction
+class Instruction implements InstructionInterface
 {
+    /**
+     * @var int
+     */
     protected $amount;
+
+    /**
+     * @var int
+     */
     protected $usageLimit;
+    
+    /**
+     * @var \DateTime
+     */
     protected $expiresAt;
 
+    /**
+     * @var int
+     */
+    protected $codeLength;
+
     public function __construct()
     {
         $this->amount = 5;
```

Comment:
Should be initialized on field creation (here https://github.com/Sylius/Sylius/pull/3339/files#diff-3755ef7e22baedb6cc72c6d1d49a3d63L21).